cpp-terminal/prompt.hpp

```diff
@@ -4,6 +4,42 @@
 #include <functional>
 
 namespace Term {
+/* Basic prompt */
+
+// indicates the results of prompt_blocking() and prompt_non_blocking
+enum class Result {
+    YES,
+    NO,
+    ERROR,
+    NONE,
+    ABORT,
+    INVALID,
+};
+// indicates the results of prompt_simple()
+enum class Result_simple { YES, NO, ABORT };
+
+// A simple yes/no prompt, requires the user to press the ENTER key to continue
+// The arguments are used like this: 1 [2/3]4 <user Input>
+Result prompt_blocking(std::string message,
+                       std::string first_option,
+                       std::string second_option,
+                       std::string prompt_indicator);
```

Comment:
I think you want to do `const std::string &message` and the same for all the other arguments also.

Reply:
The problem with copy by reference is that you cant pass a value directly then, like

```
void test(std::string test);

void test_ref(std::string& test);

int main() {

// without COR
test("STRING");
std::string test_s = "test";
test(test_s);

// with COR
test_ref("STRING");                  // <- This will fail because the const char* is not a variable that can be referenced
test_ref(test_s);                        // <- This works because the variable can be referenced
}
```

But i'll definitely check on that when I added the missing features.

Reply:
> The problem with copy by reference is that you cant pass a value directly

You can. Here is an example:
```c++
#include <string>

void f1(std::string x) {
}

void f2(const std::string &x) {
}

int main() {
    f1("ok");
    f2("ok");
    return 0;
}
```
Notice the `const`. If you don't put `const`, then indeed you get a compiler error.

Reply:
Oh, i didn't knew that const makes such a difference here. Will do that!